### ISSUE TITLE
type 'int' is not a subtype of type 'double?' in type cast #344

### ISSUE THREAD

**Issue by aquadesk** [heftapp/graphql_codegen#156]
Hi, 

I have a field `rating` in one of my model and it is `Float` type

After code gen, I have this code parsing a value from the field in one of `fromJson` functions.
```
        rating: (l$rating as double?),
```

It gives `type 'int' is not a subtype of type 'double?' in type cast` error when the value is an integer such as `3`
It works fine when there is a precision value such as `3.1` 

Thank you!

```
 factory Fragment$ProductReviewItem$attributes.fromJson(
      Map<String, dynamic> json) {
      .....
    final l$rating = json['rating'];
    return Fragment$ProductReviewItem$attributes(
       ....
        rating: (l$rating as double?),

```

```
 flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm,
    locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK
    version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.69.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!
```

**Comment by aquadesk** [comment]
more context https://github.com/gql-dart/gql/issues/344

**Comment by budde377** [comment]
Thank you for reporting this and testing out the beta. I've released `0.11.0-beta.3` which should fix this.
